### ISSUE TITLE
@serenity-js/cucumber/register collides with tsconfig-paths/register

### ISSUE THREAD

**Issue by abhinaba1080** [serenity-js/serenity-js#354]
Hi @jan-molak ,

I am seeing that Serenity is throwing error while using both:

```ts
require('@serenity-js/cucumber/register').apply(this);
require('tsconfig-paths/register');
```

Error I am getting:

```
AIBDDTests\node_modules\@serenity-js\cucumber\src\adapter.ts:12
export const adapter = cucumber.Listener({ timeout: serenity.config.stageCueTimeout });
                                ^
TypeError: cucumber.Listener is not a function
    at Object.<anonymous> (C:\Dev Room\Projects\APIBDDTests\node_modules\@serenity-js\cucumber\src\adapter.ts:12:33)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (C:\Dev Room\Projects\APIBDDTests\node_modules\@serenity-js\cucumber\src\register.ts:1:1)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
```

When I don't use tsconfig-paths/register, i do not get the error. Now, i desperately need the tsconfig-paths, in this circumstance, can we create the json serenity reports without @serenity-js/cucumber/register ? 

I use serenity 1.x and will be grate if you suggest the solution in 1.x . 

Many thanks in advance.

**Comment by jan-molak** [comment]
Hey @abhinaba1080, could you please create a minimal repo that demonstrates this issue?

**Comment by abhinaba1080** [comment]
hi @jan-molak ,

you can test that with the bdd-trader example as well... 

first do a npm install:  npm i --save-dev tsconfig-paths

just add one more line here: https://github.com/serenity-js/tutorial-bdd-trader/blob/master/features/support/configure_serenity.ts

```
require('@serenity-js/cucumber/register').apply(this);
require('tsconfig-paths/register');
```
This will produce error like I mentioned in the issue description. 
If you remove 'require('tsconfig-paths/register');' , then it will work again. 

basically tsconfig-paths helps you to configure ts import paths and shorten it. It is very usefull. 

I feel that, @serenity-js/cucumber/register does not like tscofig-paths/resister. But why? :P

**Comment by jan-molak** [comment]
@abhinaba1080 - someone would need to debug it I suppose 😄 

I haven't used `paths` as I typically use [barrels](https://basarat.gitbooks.io/typescript/docs/tips/barrel.html) instead. 
I think in Serenity/JS v1, the Cucumber typings available at `@types/cucumber` were rather rudimentary so I had to [resort to "cheating" the type system](https://github.com/jan-molak/serenity-js/blob/master/packages/cucumber/src/listener.ts#L4) by importing some typings, using `import from`, but the actual code using the old-school `require`.
Perhaps it upsets the `tsconfig-paths`?

Bear in mind that support for path mapping came out in TypeScript roughly 2 years after Serenity/JS v1 was released.

Would you be able to check if the Cucumber adapter from Serenity/JS v2 works with `tsconfig-paths` you're using?

**Comment by abhinaba1080** [comment]
Strangely, it works fine while using it with protractor with serenity 1.x

keeping the code in onPrepare in conf.ts and it worked... But definitely not working with cucumber standalone...

```ts

  onPrepare: ()=> {
    require('ts-node/register');
    require('tsconfig-paths/register');
}
```

**Comment by jan-molak** [comment]
Ha, intriguing! Please keep me posted!

**Comment by jan-molak** [comment]
Hey @abhinaba1080 - do you believe this to still be an issue, or should we close this ticket?

**Comment by jan-molak** [comment]
Hey @abhinaba1080 - I haven't heard from you regarding this ticket for a while so I'm going to close it. Please feel free to re-open it if you believe this functionality is still required.
